src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,33 @@
+using form_builder.Models;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalifElement = page.Elements.Where(_ => !string.IsNullOrEmpty(_.Properties.OptionalIfValue) || !string.IsNullOrEmpty(_.Properties.OptionalIfNotValue));
```

Comment:
Being picky but as this is an array the variable name should be plural ie. optionalIfElements.
Also the camel case is slightly wrong